core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala

```diff
@@ -93,6 +93,8 @@ class JobProgressListener(conf: SparkConf) extends SparkListener with Logging {
 
   val retainedStages = conf.getInt("spark.ui.retainedStages", SparkUI.DEFAULT_RETAINED_STAGES)
   val retainedJobs = conf.getInt("spark.ui.retainedJobs", SparkUI.DEFAULT_RETAINED_JOBS)
+  val retainedTasks = conf.getInt("spark.ui.retainedTasks", SparkUI.DEFAULT_RETAINED_TASKS)
+  val trimTasks = conf.getBoolean("spark.ui.trimTasks", false)
```

Comment:
I don't think we need the on/off switch for the trimTasks.  None of the others have off switches so I say just leave it without for now and a user can set to something huge like 100,000,000.  We can always add an off later if needed (like set to -1). If we are worried about an int being to small we could make it a long.  I don't see anything have over 2 billions tasks at this point though.